src/sagas/repository.js

```diff
@@ -158,20 +169,52 @@ export function* getDatasetPolicy({ payload }) {
   }
 }
 
-export function* addReadersToDataset({ payload }) {
-  // TODO is this going to need a loop for the list?
-  const datasetId = payload.jobResult.id;
-  const readersList = yield select(getReaders);
+export function* addReadersDataset(datasetId, reader) {
+  const readerObject = { email: reader };
   try {
     const response = yield call(
       authPost,
-      '/api/repository/v1/datasets/' + datasetId + '/policies/readers/members',
-      readersList,
+      '/api/repository/v1/datasets/' + datasetId + '/policies/reader/members',
+      readerObject,
     );
     yield put({
       type: ActionTypes.SET_DATASET_POLICY_SUCCESS,
       dataset: { data: response },
     });
+    yield call(getDatasetPolicy, { payload: datasetId });
```

Comment:
on the backend (in jade, not sam) we re-fetch the policy and return that in the response so you don't need to do it again